pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -248,7 +249,8 @@ def _run_validation(self):
 
     def _accumulated_batches_reached(self) -> bool:
         """Determine if accumulation will be finished by the end of the current batch."""
-        return self.batch_progress.current.ready % self.trainer.accumulate_grad_batches == 0
+        agb = self.trainer.accelerator.accumulate_grad_batches or self.trainer.accumulate_grad_batches
```

Comment:
do you know, are we handling the case when both are set elsewhere?

Reply:
When the accelerator property is not `None`, the trainer property will also be set.

But we want the accelerator to take precedence in that case.

Is that what you were asking?

Reply:
yeah my bad, I thought trainer.accumulate_grad_batches is None by default but it is 1, so we wouldn't actually error if both are set and instead document the precedence.